src/content/en/updates/2018/06/payment-handler-api.md

```diff
@@ -0,0 +1,212 @@
+  await navigator.serviceWorker.register(
+    // A service worker JS file is separate
+    'service-worker.js'
+  );
+  // The website must wait for the service worker to be ready
+  // before starting interactions.
+  const registration = await navigator.serviceWorker.ready;
+  // Check if Payment Handler is available
+  if (!registration.paymentManager) return;
+
+  registration.paymentManager.userHint = 'payment-handler user hint';
+  registration.paymentManager.instruments.set(
+    // Payment instrument key can be any string.
+    "https://bobpay.xyz",
+    // Payment instrument detail
+    {
+      name: 'Payment Handler Example',
+      method: 'https://bobpay.xyz/pay'
+    }
+  )
+}
+```
+
+### Receive `paymentrequest` events and get user consent for payment
+
+To handle actual payment requests, wait for `paymentrequest` events in the service 
+worker to open a window (Chrome Custom Tab) and return a payment credential 
+after getting the user's authorization for a payment.
+
+```
+const origin = 'https://bobpay.xyz';
+const methodName = `${origin}/pay`;
+const checkoutURL = `${origin}/checkout`;
+let resolver;
+let payment_request_event;
+
+self.addEventListener('paymentrequest', e => {
+  // Preserve the event for future use
+  payment_request_event = e;
+  // You'll need a polyfill for `PromiseResolver`
+  // As it's not implemented in Chrome yet.
+  resolver = new PromiseResolver();
+
+  e.respondWith(resolver.promise);
+  e.openWindow(checkoutURL).then(client => {
+    if (client === null) {
+      resolver.reject('Failed to open window');
+    }
+  }).catch(err => {
+    resolver.reject(err);
+  });
+});
+
+self.addEventListener('message', e => {
+  console.log('A message received:', e);
+  if (e.data === "payment_app_window_ready") {
+    sendPaymentRequest();
+    return;
+  }
+
+  if (e.data.methodName === methodName) {
+    resolver.resolve(e.data);
+  } else {
+    resolver.reject(e.data);
+  }
+});
+
+const sendPaymentRequest = () => {
+  if (!payment_request_event) return;
+  clients.matchAll({
+    includeUncontrolled: false,
+    type: 'window'
+  }).then(clientList => {
+    for (let client of clientList) {
+      client.postMessage(payment_request_event.total);
+    }
+  });
+}
+```
+
+### Identifying a payment app
+
+To identify the payment app from a URL-based payment method identifier (e.g., 
+`https://bobpay.xyz/pay`), you'll need to include the following declarative 
+materials.
+
+1. [A payment method 
+   identifier](https://w3c.github.io/payment-method-id/#dfn-payment-method-identifiers): 
+   points to a payment method manifest.
+1. [A payment method manifest](https://w3c.github.io/payment-method-manifest/): 
+   points to a web app manifest and supported origins.
+1. [A web app manifest](https://developer.mozilla.org/en-US/docs/Web/Manifest): 
+   describes a website that hosts a service worker that handles payment 
+   requests.
+
+[To learn more about how to implement a payment app, read this 
```

Comment:
Should be: "To learn more, read [Quick guide to implementing a payment app with the Payment Handler API](link)"

The link title should be the title of the document that you want them to read. Makes skimming easier and improves search-ability of docs.